FirebaseVertexAI/Sources/VertexAI.swift

```diff
@@ -149,12 +149,21 @@ public class VertexAI {
 
   private let auth: AuthInterop?
 
-  /// A map of active  `VertexAI` instances keyed by the `FirebaseApp` name and the `location`, in
-  /// the format `appName:location`.
-  private static var instances: [String: VertexAI] = [:]
-
-  /// Lock to manage access to the `instances` array to avoid race conditions.
-  private static var instancesLock: os_unfair_lock = .init()
+  #if compiler(>=6)
```

Comment:
Is this a pattern that you expect we'll need to use a lot across the repo? If so, potentially we could write a macro to add `nonisolated(unsafe)` (note: this suggestion is totally out of scope for this PR)

Reply:
Maybe, but we'll be able to drop Swift 5 in a few months - so probably not worth it.